src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -529,85 +527,52 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
             }
             else
             {
-                for (var i = 0; i < values.Length; i++)
+                if (values.Length == 1 && values[0].Prefix == "" &&
```

Comment:
@NTaylorMullen `SingleOrDefault()` will `throw` if there are more than one elements. Definitely not what we want here. (Further @ajaybhargavb's code is pretty clear and we've got an array so it's already efficient.)